### PR TITLE
added migration guide

### PR DESCRIPTION
**Pull request by crapStone**
Here is a little migration guide I threw together while migrating my own app.
I hope this is the right way to go, since I'm in the middle of the migration I wasn't able to test it.

This PR will remain a draft until I complete the migration to check the instructions and get feedback.

### CODE REVIEW

**Comment by nothingismagick** [comment]
Thanks for the great initiative and the good start here. Please add:

```
"@tauri-apps/api": "^1.0.0-beta-rc.3"
```

**Comment by crapStone** [comment]
I finished the migration and added instructions for updating the config file and npm and cargo "configs".

Please look through it and tell me if I should clarify something or if something is missing. My English is not the best and I got all the information on how to change these config files by creating a new Tauri project and comparing the new files to my old ones.

File: docs/en/usage/guides/migration.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
++ "@tauri-apps/cli": "^1.0.0-beta-rc.4"
+```
+
+Next update your `Cargo.toml`:
+
+- add `tauri-build` as a new build-dependency, e.g.:
+
+  ```toml
+  [build-dependencies]
+  tauri-build = { version = "1.0.0-beta-rc.0" }
```
Comment (lucasfernog) [inline docs/en/usage/guides/migration.md]: also remove winres :)

File: docs/en/usage/guides/migration.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+  ```
+
+- update the version of `tauri` to e.g. `1.0.0-beta-rc.4`
+- remove all old features of the `tauri` dependency
+- remove all features, that tauri added and add `custom-protocol` as a new one:
+  
+  ```diff
```
Comment (lucasfernog) [inline docs/en/usage/guides/migration.md:32]: this diff block isn't closed

File: docs/en/usage/guides/migration.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+}).catch(error => {
+  console.error(error);
+});
+```
+
+Complete the following steps to migrate your code:
+
+- replace all `promisified`-calls with `invoke`-calls
+- extract the `cmd` attribute of the argument object as first parameter (you
+  may have to rename it)
```
Comment (lucasfernog) [inline docs/en/usage/guides/migration.md]:
```suggestion
  may have to rename it to `snake_case` as the `cmd` is now the command function name in Rust)
```

**Comment by lucasfernog** [comment]
Amazing guide! Only missing piece is the @tauri-apps/api mention (previously it was imported as `import something from tauri/api`).

File: docs/en/usage/guides/migration.md (hunk 2 above)
Comment (crapStone) [reply docs/en/usage/guides/migration.md:32]: I should not work when I'm tired XD

**Pull request merged**